File: src/b1happynumber/Java/HappyNumberHelper.java
import java.util.HashSet;

public class HappyNumberHelper {

    // Helper function that calculates the sum of squared digits
    private static int sumOfSquaredDigits(int number) {
        int totalSum = 0;
        while (number > 0) {
            int digit = number % 10;
            number /= 10;
            totalSum += digit * digit;
        }
        return totalSum;
    }

    // Function to check if a number is a happy number
    public static boolean isHappyNumber(int n) {
        HashSet<Integer> seen = new HashSet<>();

        while (n != 1 && !seen.contains(n)) {
            seen.add(n);
            n = sumOfSquaredDigits(n);
        }

        return n == 1;
    }
}

File: src/b1happynumber/Python/happy_number_helper.py
def sum_of_squared_digits(number):
    total_sum = 0
    while number > 0:
        digit = number % 10
        number //= 10
        total_sum += digit * digit
    return total_sum


def is_happy_number(n):
    seen = set()
    while n != 1 and n not in seen:
        seen.add(n)
        n = sum_of_squared_digits(n)
    return n == 1
